### PR TITLE
Fix compatibility issue with Newtonsoft.Json 10

### PR DESCRIPTION
**Pull request by hovsepm**
Fix for https://github.com/Azure/azure-sdk-for-net/issues/2552

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

**Comment by shahabhijeet** [comment]
@azuresdkci add to whitelist

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Comment by hovsepm** [comment]
Sing job: http://azuresdkci.cloudapp.net/view/NetCore-SDK/job/NetCore-SDK-Sign/150/

File: src/SdkCommon/ClientRuntime.Azure/ClientRuntime.Azure/JsonSerializerExtensions.cs [hunk 1]
```diff
@@ -25,7 +25,10 @@ public static class JsonSerializerExtensions
             }
             JsonSerializer newSerializer = new JsonSerializer();
             var properties = typeof(JsonSerializer).GetTypeInfo().DeclaredProperties;
-            foreach (var property in properties.Where(p => p.SetMethod != null && !p.SetMethod.IsPrivate))
+            foreach (var property in properties.Where(p =>
```
Comment (shahabhijeet) [inline src/SdkCommon/ClientRuntime.Azure/ClientRuntime.Azure/JsonSerializerExtensions.cs]: @hovsepm what if Binder property is being set by the serializingSettings?
Will be ignore in that case.
I think we should not ignore Obsolete properties, rather ignore properties that are throwing exceptions.

File: src/SdkCommon/ClientRuntime.Azure/ClientRuntime.Azure/ResourceJsonConverter.cs [hunk 2]
```diff
@@ -188,7 +188,10 @@ protected JsonSerializer GetSerializerWithoutCurrentConverter(JsonSerializer ser
             }
             JsonSerializer newSerializer = new JsonSerializer();
             var properties = typeof(JsonSerializer).GetTypeInfo().DeclaredProperties;
-            foreach (var property in properties.Where(p => p.SetMethod != null && !p.SetMethod.IsPrivate))
+            foreach (var property in properties.Where(p =>
```
Comment (shahabhijeet) [inline src/SdkCommon/ClientRuntime.Azure/ClientRuntime.Azure/ResourceJsonConverter.cs]: @hovsepm same as above

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
The right fix is that json.net is fixed.
Why suddenly Binder property starts throwing.
Because if the calling code initialize Binder property we do not see this issue.

**Comment by shahabhijeet** [comment]
@hovsepm if this fixes #2552 then this is good to go.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**